core/src/main/java/com/linecorp/armeria/server/annotation/Param.java

```diff
@@ -22,10 +22,16 @@
 import java.lang.annotation.Target;
 
 /**
- * Annotation for mapping a parameter of a request onto a method parameter.
+ * Annotation for mapping a parameter of a request onto the following elements.
+ *
+ * <p>a method parameter
+ *
+ * <p>or, a field of a request bean (See: {@link RequestConverter} and {@link RequestConverterFunction})
+ *
+ * <p>or, a setter method of a request bean (See: {@link RequestConverter} and {@link RequestConverterFunction})
  */
 @Retention(RetentionPolicy.RUNTIME)
-@Target(ElementType.PARAMETER)
+@Target({ ElementType.PARAMETER, ElementType.FIELD, ElementType.METHOD })
```

Comment:
I wonder if we should use a parameter-scope annotation for setter methods:

```java
@Param("foo")
public void setFoo(String foo) { ... }
// vs.
public void setFoo(@Param("foo") String foo) { ... }
```

I'm not sure which is better, but we could potentially inject more than one params in a single call:

```java
public void setAll(@Param("foo") String foo, @Param("bar") String bar) { .. }
```

What do you think? Which sounds better or do you think we should implement both? (not necessarily all in this PR - no pressure)

Reply:
I will implement it later.

Reply:
Done.
now `@Param`/`@Header` can be used to bean method and method params